ezyrb/ann.py

```diff
@@ -49,48 +50,48 @@ def fit(self, points, values):
         :return the training loss value at termination (after niter iterations).
         :rtype: float.
         """
-        layers = [points.shape[1], 10, 5, values.shape[1]]
         # ordered list with the number of neurons per layer (to be modified by the user)
-        # (i.e., layers[i]=number of neurons in the layer i)
-        niter=5000
+        # (i.e., layers[i] = number of neurons in the layer i)
+        layers = [points.shape[1], 10, 5, values.shape[1]]
+        niter = 5000
         arguments = []
         for i in range(len(layers)-2):
             arguments.append(nn.Linear(layers[i], layers[i+1]))
             arguments.append(nn.Tanh())
         arguments.append(nn.Linear(layers[len(layers)-2], layers[len(layers)-1]))
         arguments.append(nn.Identity())
-        self.model = nn.Sequential(*arguments)
         # ANN structural model definition
-        self.optimizer=torch.optim.Adam(self.model.parameters(), lr=0.001, betas=(0.9, 0.999), eps=1e-08, weight_decay=0, amsgrad=False)
+        self.model = nn.Sequential(*arguments)
         # setting of the optimization solver (Adam)
+        self.optimizer = torch.optim.Adam(self.model.parameters(), lr=0.001, betas=(0.9, 0.999), eps=1e-08, weight_decay=0, amsgrad=False)
         points = torch.from_numpy(points).float()
         values = torch.from_numpy(values).float()
         for epoch in range(niter):
-            y_pred = self.model(points)
             # forward propagation (net evaluation at 'points')
-            loss = self.criterion(y_pred, values)
+            y_pred = self.model(points)
             # compute the training loss
-            self.optimizer.zero_grad()
+            loss = self.criterion(y_pred, values)
             # zero the gradients
-            old_loss = loss.item() 
+            self.optimizer.zero_grad()
             # loss values extraction (type: float)
-            loss.backward()
+            old_loss = loss.item() 
```

Comment:
It looks an unused variable

Reply:
It seems to be required for the subsequent self.optimizer.step(). I tried to remove "loss.backward()" but the training is not performing as expected (the magnitude of the training loss at termination is meaningful).

Reply:
`backward` is for sure a fundamental step. I meant the `old_loss` variable.

Reply:
That's true, thanks.